src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -465,6 +466,13 @@ public void Write(ArraySegment<byte> data)
         {
             ProduceStartAndFireOnStarting().GetAwaiter().GetResult();
 
+            if (!_canHaveBody)
+            {
+                // Don't write to body for HEAD requests or 101, 204, 205, 304 responses.
+                LogNonBodyRequestWrite(data.Count);
```

Comment:
Question on whether GET responses can serve HEAD requests?

Can differentiate the response codes from method, so 101, 204, 205, 304 throw and HEAD eats?